common/cache/cache.go

```diff
@@ -104,3 +104,11 @@ type Entry interface {
 	// CreateTime represents the time when the entry is created
 	CreateTime() time.Time
 }
+
+type PutCallback interface {
+	OnPut()
+}
+
+type EvictCallback interface {
+	OnEvict()
+}
```

Comment:
These two callbacks are introduced to allow callbacks to fire based on the cache item's lifecycle events.